backend/docs/api.md

```diff
@@ -0,0 +1,151 @@
+# API
+
+## Routes
+
+### GET /shop/:shop/
+
+Returns the shop configuration.
+
+Example response:
+
+```json
+{
+  "@id": "http://noqueue.example.org/shop/cat-food/",
+  "label": "Cat food"
+}
+```
+
+### PUT /shop/:shop/
+
+Updates the shop configuration.
+
+Example request:
+
+```json
+{
+  "label": "Cat food"
+}
+```
+
+### POST /shop/:shop/ticket/
+
+Issues a new ticket.
+On success, the URL of the ticket can be found in the `location` header.
+
+Example request:
+
+```json
+{
+  "start": "2020-05-01T11:00:00Z",
+  "end": "2020-05-01T11:15:00Z"
+}
+```
+
+### GET /shop/:shop/ticket/available?start&end
+
+Returns an overview of issued tickets for the given time frame.
+`start` and `end` must be given as ISO date strings. 
+
+Example response:
+
+```json
+[{
+  "start": "2020-05-01T11:00:00Z",
+  "end": "2020-05-01T11:15:00Z",
+  "reserved": 1,
+  "allowed": 2,
+  "available": 1
+}, {
+  "start": "2020-05-01T11:15:00Z",
+  "end": "2020-05-01T12:00:00Z",
+  "reserved": 0,
+  "allowed": 2,
+  "available": 2
+}]
+```
+
+### GET /shop/:shop/ticket/:ticket
+
+Returns the data of an issued ticket.
+
+Example response:
+
+```json
+{
+  "@id": "http://noqueue.example.org/shop/cat-food/ticket/6c39f8e5-e085-49d0-a4bc-c4673768c4d2",
+  "user": "default",
+  "start": "2020-05-01T11:00:00Z",
+  "end": "2020-05-01T11:15:00Z",
+  "ready": false,
+  "used": false
+}
+```
+
+### PUT /shop/:shop/ticket/:ticket
```

Comment:
Would this be used by a scanner to validate a ticket? Does it need a special authorization header, so a user cannot re-use a ticket multiple times by updating it to `ready` again?

Reply:
I will add auth in a separate PR. At the moment everyone can update everything. Based on the feature for v1.0.0 I would change it to:

- allow only admins to change a ticket
- if check-in is supported, allow anonymous users (people that own the URL) to change the ready state